lib/rules/no-constant-condition.js

```diff
@@ -16,6 +16,27 @@ module.exports = function(context) {
     // Helpers
     //--------------------------------------------------------------------------
 
+
+    /**
+     * Checks if a branch node of LogicalExpression short circuits the whole condition
+     * @param {ASTNode} node The branch of main condition which needs to be checked
+     * @param {string} operator The operator of the main LogicalExpression.
+     * @returns {boolean} true when condition short circuits whole condition
+     */
+    function isShortCircuit(node, operator) {
+        switch (node.type) {
+            case "Literal":
+                return (operator === "||" && node.raw === "true") ||
```

Comment:
@pvamshi Yes, compare `node.value` to `true` (the boolean, not the string), that should work better.